tools/build_everything.py

```diff
@@ -146,8 +146,12 @@
         base_source_paths = options.source_dir
         if not base_source_paths:
             base_source_paths = ['.']
-        
-        all_tests = find_tests(base_source_paths[0])
+       
+        all_tests = {}
+        for platform in platforms:
+            for toolchain in toolchains:
+                platform_toolchain_tests = find_tests(base_source_paths[0], platform, toolchain)
+                all_tests.update(platform_toolchain_tests)
```

Comment:
It used to be that this was the case. The tests that were ran were always the same for every target and toolchain. But now certain boards have certain "features" that other boards don't support. There could (and should) be tests under these feature folders that should only be built/tested if the board supports this feature. So this is why test discovery is now dependent on target and toolchain.